src/v/test_utils/tests/gtest_test.cc

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#include "test_utils/test.h"
+
+#include <seastar/core/sleep.hh>
+
+/*
+ * TEST_(ASYNC|CORO)() is the Seastar-enabled equivalent of TEST().
+ */
+TEST_ASYNC(SeastarTest, Sleep) {
+    seastar::sleep(std::chrono::milliseconds(100)).get();
+}
+
+TEST_CORO(SeastarTest, SleepCoro) {
+    co_await seastar::sleep(std::chrono::milliseconds(100));
+    ASSERT_EQ_CORO(100, 100);
+}
+
+/*
+ * TEST_F_(ASYNC|CORO)() is the Seastar-enabled equivalent of TEST_F()
+ */
```

Comment:
Is there a particular reason we're mixing naming styles?

I know GoogleTest uses an _ to join the generated test class and test name, this recommends using PascalCase names, should we stick to that advice?

Reply:
From Googletest docs:

TEST() arguments go from general to specific. The first argument is the name of the test suite, and the second argument is the test’s name within the test suite. Both names must be valid C++ identifiers, and they should not contain any underscores (_). A test’s full name consists of its containing test suite and its individual name. Tests from different test suites can have the same individual name.

Source: http://google.github.io/googletest/primer.html#simple-tests

Reply:
Are you referring to `struct MySeastarFixture : public seastar_test`? Yeh, I assume that everyone is going to continue to use lower case snake case naming, but good point about standardizing. Is it just about aesthetics: like pascal case just looks better the way that google test uses the names?

Reply:
Yes this comment is about naming convention. We should standardize 😄 

It's not *just* aesthetics. Check out: https://clang.llvm.org/extra/clang-tidy/checks/google/readability-avoid-underscore-in-googletest-name.html

Reply:
https://google.github.io/googletest/faq.html#why-should-test-suite-names-and-test-names-not-contain-underscore